File: src/main/java/com/aitherapist/aitherapist/scheduled/IScheduledNotificationRepository.java
package com.aitherapist.aitherapist.scheduled;

import com.aitherapist.aitherapist.domain.enums.NotificationStatus;
import com.aitherapist.aitherapist.scheduled.ScheduledNotification;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;
import java.util.List;

public interface IScheduledNotificationRepository extends JpaRepository<ScheduledNotification, Long> {
    List<ScheduledNotification> findByTelegramChatId(Long chatId);
    List<ScheduledNotification> findByStatusAndTriggerTimeAfter(NotificationStatus status, LocalDateTime triggerTime);
}

File: src/main/java/com/aitherapist/aitherapist/scheduled/IScheduledNotificationRepository.kt
package com.aitherapist.aitherapist.telegrambot.scheduled

import com.aitherapist.aitherapist.scheduled.NotificationStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime

interface IScheduledNotificationRepository : JpaRepository<ScheduledNotification, Long> {
    fun findByTelegramChatId(chatId: Long): List<ScheduledNotification>
    fun findByStatusAndTriggerTimeAfter(status: NotificationStatus, triggerTime: LocalDateTime): List<ScheduledNotification>
}
